### DISCUSSION TITLE
Create application with api gateway but is required the listener

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#403] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12.0.2

### Description

I'm trying to create a new application that will create a new api gateway but the first step is blocked as it is required a listener, but you can create listeners only after creating the api gateway.

![image](https://github.com/mia-platform/community/assets/18056946/ab4c585c-774c-41bc-b6ab-00e4c69eab56)

### Actual Outcome

The application creation flow is blocked

### Expected Outcome

I'm able to create the application

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @elmalakomar does this happen in a self-hosted console or our SaaS? 

If the first is your scenario, it may be that you did not run the marketplace updater installation script; it will update the application to ship listeners as well as the Envoy api gateway.

Otherwise you need to create the Envoy API Gateway first, setup your listener and then create the application linking the existing API gateway.

Sadly this is not a bug we can fix, but more a friction we are forced to give to prevent inconsistent configuration definition. The copy in the error badge could be improved to be more explicit.
